iceoryx-sys/src/publisher_options.rs

```diff
@@ -6,7 +6,7 @@ use crate::ConsumerTooSlowPolicy;
 
 use std::marker::PhantomData;
 
-pub(super) struct PublisherOptions {
+pub struct PublisherOptions {
     pub history_capacity: u64,
```

Comment:
Out of curiosity, what happens when we make the history capacity an `uint16_t` in cpp and forget to adjust it here?

Reply:
There is a layer of indirection
```cpp
let raw = cpp!([service as "const char *",
                instance as "const char *",
                event as "const char *",
                history_capacity as "uint64_t", // <- this comes from the Rust struct
                node_name as "const char *",
                offer_on_create as "bool",
                subscriber_too_slow_policy as "uint8_t"]
                -> *mut Publisher as "PublisherPortUser*"
{
    PublisherOptions options; // <- this is the C++ struct
    options.historyCapacity = history_capacity;
    options.nodeName = IdString_t(TruncateToCapacity, node_name);
    options.offerOnCreate = offer_on_create;
    options.subscriberTooSlowPolicy = static_cast<ConsumerTooSlowPolicy>(subscriber_too_slow_policy);
    auto portData = PoshRuntime::getInstance().getMiddlewarePublisher(
        {
            IdString_t(TruncateToCapacity, service),
            IdString_t(TruncateToCapacity, instance),
            IdString_t(TruncateToCapacity, event)
        },
        options
    );
    return new PublisherPortUser(portData);
});
```
Depending on the build flags, this will be a warning or an error.